### PR TITLE
Remove name parameter from animation directive.

### PR DESCRIPTION
**Pull request by gspencergoog**
This PR removes the required first argument for the `{@animation...}` directive, making it into an optional argument.

It also builds a way to parse arguments for directives that is more consistent, and leverages the Dart arg parser.  The arguments may be quoted in order to contain spaces if needed. Updates _injectExamples to use this parser as well as the _injectAnimations function.

The old animation directive form is still supported, but will give a deprecation warning if used.  I also added a new type of warning for this: `PackageWarning.deprecated`.

### CODE REVIEW

File: lib/src/model.dart [hunk 1]
```diff
@@ -3718,29 +3741,69 @@ abstract class ModelElement extends Canonicalization
...
+    // so that they parse as options.
+    final Iterable<String> args = matches.map<String>((Match match) {
+      String option = '';
+      if (match[1] != null) {
+        option = '--${match[1]}';
+      }
+      return option + (match[3] ?? '') + (match[4] ?? '');
+    });
+
+    return argParser.parse(args);
```
Comment (jcollins-g) [inline lib/src/model.dart]: I think ArgParserExceptions thrown here probably should not be fatal and should instead emit some sort of helpful warning.  You might have to carry along some extra data into _parseArgs to make that useful without reflection.

File: lib/src/model.dart [hunk 2]
```diff
@@ -3554,87 +3554,110 @@ abstract class ModelElement extends Canonicalization
...
+      final ArgParser parser = new ArgParser();
+      parser.addOption('id');
+      final ArgResults args = _parseArgs(basicMatch[1], parser);
+      final List<String> positionalArgs = args.rest.sublist(0);
+      String uniqueId;
+      bool wasDeprecated = false;
+      if (positionalArgs.length == 4) {
+        // Supports the original form of the animation tag for backward
+        // compatibility.
+        uniqueId = positionalArgs.removeAt(0);
```
Comment (jcollins-g) [inline lib/src/model.dart]: This will of course completely discard the name.  Given that anchors do not actually work yet I suppose this is probably harmless?

File: testing/test_package/lib/example.dart [hunk 3]
```diff
@@ -357,16 +357,48 @@ class Dog implements Cat, E {
...
+  /// Animation method with quoted name
+  ///
+  /// {@animation 100 100 http://host/path/to/video.mp4 id="quotedNamedAnimation"}
+  /// {@animation 100 100 http://host/path/to/video.mp4 id='quotedNamedAnimation2'}
+  /// More docs
+  void withQuotedNamedAnimation() {}
+
+  /// Animation method with invalid name
+  ///
+  /// {@animation 100 100 http://host/path/to/video.mp4 id=2isNot-A-ValidName}
```
Comment (jcollins-g) [inline testing/test_package/lib/example.dart]: Suggest adding test cases for `{@animation 250 250 id=outOfOrder http://host/path/to/video.mp4}`
and `{@animation 149 150 http://host/path/to/video.mp4 notId=unexpected}` to validate more edge case behavior

Comment (gspencergoog) [reply testing/test_package/lib/example.dart]: OK, added.

File: lib/src/model.dart (hunk 2 above)
Comment (jcollins-g) [reply lib/src/model.dart]: Right, got it.

**Comment by jcollins-g** [comment]
I think github's commenting system doesn't like it when commits are replaced on the branch rather than added as the review takes place.  github doesn't seem to want to offer anything except an emoji response to your reply on ArgParserException and throws some errors because it can't find the previous commit that was attached to.
